Use check-types over type-check

arbourd opened the pull request:
We've decided to move on from this package and would love to use
something a little bit more generally available.

arbourd commented:
https://travis-ci.com/github/arbourd/rsxjs/builds/169679673

karimsa commented on src/channel/channel.ts:
```diff
@@ -70,8 +70,8 @@ export class Channel<T> implements IChannel<T> {
-    typeCheck('object', config)
-    typeCheck('number?', config.bufferSize)
+    check.assert.object(config)
+    check.assert.number(config.bufferSize)
```
I believe this is the wrong type. The bufferSize is optional for this class, since the channel can toggle between buffered and non-buffered. So the check-types equivalent should be: `check.assert.maybe.number`.

Commit:
Use check-types over type-check

We've decided to move on from this package and would love to use
something a little bit more generally available.

arbourd commented:
Not sure why the actions didn't kick in because I did rebase on master 🤷‍♂️😅

karimsa commented:
Thanks for catching that! That was my bad. I forgot to include PRs in the trigger list. Can you try pulling master on top of your branch or rebasing?

arbourd commented:
Cool, gots it.

karimsa commented:
Released as [v0.6.0](https://github.com/karimsa/rsxjs/releases/tag/v0.6.0).

arbourd commented:
Thanks @karimsa! Appreciate the quick turn around 😁
